staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher_whitebox_test.go

```diff
@@ -686,7 +686,9 @@ func TestCacherNoLeakWithMultipleWatchers(t *testing.T) {
 			case <-stopCh:
 				return
 			default:
+				cacher.Lock()
```

Comment:
the drawback is that now low level test need lock before accessing `bookmarkWatchers` but we could change some method names to highlight is is not safe

Reply:
I see you already did that - this looks fine.